mkdocs_redirects/plugin.py

```diff
@@ -2,26 +2,51 @@
 Copyright 2019-2022 DataRobot, Inc. and its affiliates.
 All rights reserved.
 """
-import logging
+from __future__ import annotations
+
 import os
 import posixpath
+from typing import TypedDict
 
 from mkdocs import utils
 from mkdocs.config import config_options
-from mkdocs.plugins import BasePlugin
+from mkdocs.plugins import BasePlugin, get_plugin_logger
 from mkdocs.structure.files import File
 
-log = logging.getLogger("mkdocs.plugin.redirects")
+log = get_plugin_logger(__name__)
+
 
+def gen_anchor_redirects(anchor_list: list):
+    """
+    Generate a dictionary of redirects for anchors.
 
+    :param anchor_list: A list of tuples containing old anchors and new links.
+    :return: A string of JavaScript redirects for the anchors.
```

Comment:
nit: We don't generate docs so the docstring style is not important, but I'd favor Google-style docstrings rather than Sphinx-style ones, since this is a MkDocs plugin :yum: